packages/ember-htmlbars/lib/keywords/mount.js

```diff
@@ -0,0 +1,179 @@
+      params.length === 1
+    );
+
+    assert(
+      'The first argument of {{mount}} must be quoted, e.g. {{mount "chat-engine"}}.',
+      typeof name === 'string'
+    );
+
+    assert(
+      'You used `{{mount \'' + name + '\'}}`, but the engine \'' + name + '\' can not be found.',
+      env.owner.hasRegistration(`engine:${name}`)
+    );
+
+    let engineInstance = env.owner.buildChildEngineInstance(name);
+
+    engineInstance.boot();
+
+    let state = {
+      parentView: env.view,
+      manager: prevState.manager,
+      controller: lookupEngineController(engineInstance),
+      childOutletState: childOutletState(name, env)
+    };
+
+    setOwner(state, engineInstance);
+
+    return state;
+  },
+
+  childEnv(state, env) {
+    return buildEnvForEngine(getOwner(state), env);
+  },
+
+  isStable(lastState, nextState) {
+    return isStable(lastState.childOutletState, nextState.childOutletState);
+  },
+
+  isEmpty(/* state */) {
+    return false;
+  },
+
+  render(node, env, scope, params, hash, template, inverse, visitor) {
+    let state = node.getState();
+
+    let engineInstance = getOwner(state);
+
+    let engineController = lookupEngineController(engineInstance);
+
+    let engineTemplate = lookupEngineTemplate(engineInstance);
+
+    let options = {
+      layout: null,
+      self: engineController
+    };
+
+    let engineEnv = buildEnvForEngine(engineInstance, env);
+
+    let nodeManager = ViewNodeManager.create(node, engineEnv, hash, options, state.parentView, null, null, engineTemplate);
+
+    state.manager = nodeManager;
+
+    nodeManager.render(engineEnv, hash, visitor);
+  },
+
+  rerender(node, env, scope, params /*, hash, template, inverse, visitor */) {
+    let controller = node.getState().controller;
+    if (controller) {
+      let model = read(params[1]);
+      controller.set('model', model);
+    }
+  }
+};
+
+function childOutletState(name, env) {
+  let topLevel = env.view.ownerView;
+  if (!topLevel || !topLevel.outletState) { return; }
+
+  let outletState = topLevel.outletState;
+  if (!outletState.main) { return; }
+
+  let selectedOutletState = outletState.main.outlets['__ember_orphans__'];
+  if (!selectedOutletState) { return; }
+  var matched = selectedOutletState.outlets[name];
+  if (matched) {
+    var childState = new EmptyObject();
+    childState[matched.render.outlet] = matched;
+    matched.wasUsed = true;
+    return childState;
+  }
+}
+
+function isStable(a, b) {
+  if (!a && !b) {
+    return true;
+  }
+  if (!a || !b) {
+    return false;
+  }
+  for (var outletName in a) {
+    if (!isOutletStable(a[outletName], b[outletName])) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function lookupEngineController(engineInstance) {
+  return engineInstance.lookup('controller:application');
+}
+
+function lookupEngineView(engineInstance, ownerView) {
+  let engineView = engineInstance.lookup('view:application') || engineInstance.lookup('view:toplevel');
```

Comment:
We should no longer look for `view:application` (support for using/looking up a `view` has been removed in canary).